R/parse_params.R

```diff
@@ -35,78 +36,156 @@ parse_params <- function(station_id, start, end, hour = FALSE) {
     station_id <- "*"
   }
 
-  # Parse Dates -------------------------------------------------------------
+  # Set Date Defaults ---------------------------------------------------------
 
-  if(hour) {
+  ## Variable dictionary ##################
+  # end/start: user input used for conditionals, character
+  # end_/start_: initial defaults, character
+  # end_parsed/start_parsed: used for messages and validation, datetime
+  # start_f/time_interval: what is actually passed to the API
+  #########################################
+
+  if (is.null(end)) {
+    if(isTRUE(hour)) {
+      end_ <- format( #keep as character until later, for consistency
```

Comment:
`end_` here, as defined, relates to my comment on `az_hourly.R`. Does not subtracting an hour make sense for not returning a two-hour window?

Reply:
This _might_ be able to be changed to the current hour.  Not entirely sure where it's getting used anymore.

Reply:
This did end up being part of the fix. Thanks for spotting it!